Assets/Scripts/WeaponInventory.cs

```diff
@@ -53,8 +61,16 @@ public void EquipWeapon(WeaponId weaponId)
             // Set ammo to infinite for now
             switch (weaponId)
             {
+                case WeaponId.VETO:

```

Comment:
instead of doing a switch here we should have a set of maps that map from weaponId to the weapon attributes

'
struct WeaponAttributes
{
    int MaxAmmo;
    float CoolDown;
}
'

' WeaponAttribute WeaponAttributeMap[ 8 ] = {/* weapon attributes for each weapon go here. i.e. the first slot will map to fists, the next slot make to veto, then magic bullet etc... */} '